caluma/caluma_workflow/domain_logic.py

```diff
@@ -0,0 +1,96 @@
+from django.core.exceptions import ValidationError
+
+from caluma.caluma_core.events import send_event
+from caluma.caluma_user.models import BaseUser
+
+from ..caluma_form.models import Document
+from . import events, models, utils
+
+
+class StartCaseLogic:
+    @staticmethod
+    def validate_for_start(data):
+        form = data.get("form")
+        workflow = data.get("workflow")
+
+        if form:
+            if (
+                not workflow.allow_all_forms
+                and not workflow.allow_forms.filter(pk=form.pk).exists()
+            ):
+                raise ValidationError(
+                    f"Workflow {workflow.pk} does not allow to start case with form {form.pk}"
+                )
+
+        return data
+
+    @staticmethod
+    def pre_start(validated_data, user):
```

Comment:
I'd like to see better naming: What does this method *do*?

Reply:
Also, some short docstring would be nice, even if this is not "public API"

Reply:
Maybe it'd make sense to explain *why* we're doing this 3-step dance. Could be a docstring on the `StartCaseLogic` class, for example. Just so future readers will know

Reply:
What? I thought we discussed this naming through yesterday! :disappointed:

Reply:
I think `pre_start` and `post_start` are acceptable method names. I agree though that a docstring for `StartCaseLogic` might help, maybe something along the lines of

> Shared domain logic for starting cases.
>
> Used in the `SaveCase` mutation and in the `start_case` api. The logic for case creation is split in two parts (`pre_start` and `post_start`) so that in between the appropriate `create` method can be called (the serializer calls `super.create`).

Reply:
Done